### DISCUSSION TITLE
Ford Transit Campervan Conversion Project Using MicroPython and ESP32

### DISCUSSION THREAD

**Discussion by messydeer** [micropython/micropython#11567] [Show and tell]
Hi!

I am converting a 2020 Ford Transit van into a campervan. I would like to learn how to program an ESP32 with micropython to monitor and control my systems in the van. Here's an overview:

1. Electrical

* Two Solaria PowerXT-400R-PM solar panels on the roof
* Victron mppt-tr 150/85 solar charge controller 
* Victron lynx distributor
* Victron multiplus charger/inverter
* Two 30 amp Victron Orion Tr Smart isolated dc/dc chargers
* Two parallel wired 12volt LiFePO4 batteries totalling500Ah. 
* 12v and 110v outlets and devices including:

** interior and exterior lighting
** air conditioning
** air heating
** running hot and cold water
** an electrical cooktop, microwave/convection oven, and refrigerator/freezer. 
** video security system

2. Plumbing

* Under floor mounted fresh and grey water tanks with level monitors and remote valve actuators
* Recirculating shower
* Kitchen sink

3. Mechanical

* gasoline (from fuel tank) fired or electric heat
* air conditioning

ChatGPT suggested I use an ESP32 to monitor and control these systems. Because I am a beginner I have chosen to learn MicroPython via Thonny, based on ChatGPT recommendations. I have bought a Freenove Wrover starter kit and am going through their tutorial.

 I have found the learning curve fairly steep, but I've only taken the first few steps. Any suggestions on how to go about this process will be helpful, including eBooks and other resources. I want to focus mainly on what will help me complete my project.

Thanks,
Dan

**Comment by 2dof** [comment]
Do not invent something if already existing bu if You want do something yourself  use for example RaspbberyPi as main central computer  (+ monitor with touchpad) to gather all data from all modules ( i.e solar system control unit, power management  etc. they  usually have bt/wi-fi and application for smartphone/computer) , or instal some scada and create your own real-time-control/monitor application. 

Do not listen GBT (it is generative language model not inductive system).

**Comment by brave-ulysses** [comment]
interesting project. might look at openauto and crankshaft for some ideas. expedition portal may have some builds...

good luck!
